Json parser for OrderedDict type

antonplietzsch opened the pull request:
The functions `read_powergrid` and `write_power_grid` did not work for the new `OrderedDict` type of nodes and lines. I added some dipatches in JsonParser.jl, such that both `Array` and `OrderedDict` are supported.

SabineAuer commented:
@antonplietzsch : great job! Can you also add a test for the extension of these functions?

antonplietzsch commented:
1. I fixed a bug that appeared in the tests: The parser was not able to correctly read old .json files, since they didn't have a name field for the components. I changed the code such that a name field is only written for grids in `OrderedDict` type. For the `Array` type the `write_powergrid` function is the same as before.
2. I added tests for reading and writing of grids in `OrderedDict` type.
